sys/androidmedia/gstamcvideodec.h

```diff
@@ -102,6 +102,7 @@ struct _GstAmcVideoDec
   GMutex srccaps_lock;
   GCond srccaps_cond;
   gboolean srccaps_set;
+  gboolean waiting_segment;
```

Comment:
You're missing the initialization of this class variable in the class init function.